uefi/src/data_types/strs.rs

```diff
@@ -55,6 +91,23 @@ pub enum FromStrWithBufError {
     BufferTooSmall,
 }
 
+impl Display for FromStrWithBufError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                Self::InvalidChar(_usize) => "invalid character",
+                Self::InteriorNul(_usize) => "encountered null character",
```

Comment:
For the interior null error, maybe something like `"interior null character at index {index}"`, to clarify that it's specifically an _interior_ null that is the problem (a trailing null is expected and required).